src/server/server_impl.go

```diff
@@ -52,6 +54,37 @@ func (server *server) AddDebugHttpEndpoint(path string, help string, handler htt
 	server.debugListener.endpoints[path] = help
 }
 
+// add an http/1 handler at the /json endpoint which allows this ratelimit service to work with
+// clients that cannot use the gRPC interface (e.g. lua)
+// example usage from cURL with domain "dummy" and descriptor "perday":
+// echo '{"domain": "dummy", "descriptors": [{"entries": [{"key": "perday"}]}]}' | curl -vvvXPOST --data @/dev/stdin localhost:8080/json
+func (server *server) AddJsonHandler(svc pb.RateLimitServiceServer) {
+	handler := func(writer http.ResponseWriter, request *http.Request) {
+		var req pb.RateLimitRequest
+
+		if err := json.NewDecoder(request.Body).Decode(&req); err != nil {
+			logger.Warnf("error: %s", err.Error())
+			http.Error(writer, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		resp, err := svc.ShouldRateLimit(nil, &req)
+		if err != nil {
+			logger.Warnf("error: %s", err.Error())
+			http.Error(writer, err.Error(), http.StatusBadRequest)
+			return
+		}
+		logger.Debugf("resp:%s", resp)
+		if resp.OverallCode == pb.RateLimitResponse_OVER_LIMIT {
+			http.Error(writer, "over limit", http.StatusTooManyRequests)
+		} else if resp.OverallCode == pb.RateLimitResponse_UNKNOWN {
+			http.Error(writer, "unknown", http.StatusInternalServerError)
+		}
```

Comment:
When does this happen? Can this happen?

Reply:
I added this check for completeness with the protobuf definition.  I have never personally seen it.